packages/web-components/src/components/notice-choice/notice-choice.ts

```diff
@@ -0,0 +1,526 @@
+        </div>
+        ${this.getBpidLegalText()}
+    </section>`;
+  }
+  protected emailChanged(email) {
+    if (this.changed === false) {
+      checkPreferencesv3(email).then((response) => {
+        if (
+          response === 'S' &&
+          JSON.stringify(this.values) !==
+            JSON.stringify({
+              ...this.values,
+              EMAIL: false,
+            })
+        ) {
+          this.fetchedPref = email;
+          this.values = {
+            ...this.values,
+            EMAIL: false,
+          };
+          this._onChange('NC_HIDDEN_EMAIL', null);
+        }
+      });
+    }
+  }
+  protected _getOptionByQuestion = (question, OptInContentValue) => {
+    let OptInContent = OptInContentValue;
+    if (!OptInContent) {
+      OptInContent = this.ncData.OptInContent;
+    }
+    let option;
+    switch (question) {
+      case 'EMAIL': {
+        option = OptInContent.fourQuestionApp[0];
+        break;
+      }
+      case 'PHONE': {
+        option = OptInContent.fourQuestionApp[1];
+        break;
+      }
+      case 'POSTAL': {
+        option = OptInContent.fourQuestionApp[2];
+        break;
+      }
+      default: {
+        option = OptInContent.fourQuestionApp[0];
+        break;
+      }
+    }
+    return option;
+  };
+  protected _buildCheckboxes(OptInContent: any) {
+    const fieldElements: any = {};
+    const fieldCollections = {
+      EMAIL: {
+        id: 'EMAIL',
+        labelText: this._getOptionByQuestion('EMAIL', OptInContent)
+          .optionTextPost,
+      },
+      PHONE: {
+        id: 'PHONE',
+        labelText: this._getOptionByQuestion('PHONE', OptInContent)
+          .optionTextPost,
+      },
+      POSTAL: {
+        id: 'POSTAL',
+        labelText: this._getOptionByQuestion('POSTAL', OptInContent)
+          .optionTextPost,
+      },
+    };
+    if (this.questionchoices) {
+      // by email
+      if (this.questionchoices.indexOf('1') > -1) {
+        fieldElements.EMAIL = fieldCollections.EMAIL;
+      }
+      // by Phone
+      if (this.questionchoices.indexOf('2') > -1) {
+        fieldElements.PHONE = fieldCollections.PHONE;
+      }
+      // by Postal mail
+      if (this.questionchoices.indexOf('3') > -1) {
+        fieldElements.POSTAL = fieldCollections.POSTAL;
+      }
+    }
+    return fieldElements;
+  }
+
+  _onChange(field: string, value: string | null) {
+    const pwsFieldsMap = {
+      NC_HIDDEN_EMAIL: 'permission_email',
+      NC_HIDDEN_PHONE: 'permission_phone',
+      NC_HIDDEN_POSTAL: 'permission_postal',
+    };
+    if (Object.prototype.hasOwnProperty.call(pwsFieldsMap, field)) {
+      field = pwsFieldsMap[field];
+    }
+    const init = {
+      bubbles: true,
+      detail: {
+        field,
+        value: pwsValueMap(value),
+      },
+    };
+    this.dispatchEvent(
+      new CustomEvent(`${ddsPrefix}-notice-choice-change`, init)
+    );
+  }
+}
+/* @__GENERATE_REACT_CUSTOM_ELEMENT_TYPE__ */
```

Comment:
@sanjitbauli This can be removed. There's already a React version and this is interfering with the `react-wrapper` builds since it's not properly set up for the wrappers. Once this is removed it should pass checks and be ready to get merged.

Reply:
@kennylam done.